### PR TITLE
Fix contact release

### PR DESCRIPTION
**Pull request by norkans7**
Fixes https://github.com/rapidpro/rapidpro/issues/1611

### CODE REVIEW

File: temba/contacts/migrations/0150_full_release_deleted_contacts.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+from django.db import migrations
+
+
+def full_release_deleted_contacts(apps, schema_editor):  # pragma: no cover
+    Contact = apps.get_model("contacts", "Contact")
+
+    deleted_contacts = Contact.objects.filter(is_active=False, name=None)
+
+    for contact in deleted_contacts:
```
Comment (norkans7) [inline temba/contacts/migrations/0150_full_release_deleted_contacts.py:11]: copied over the method https://github.com/nyaruka/rapidpro/pull/3689/files#diff-7b90ea5e3fb9ba05a8e735870b6aca87d34d2c41584165c9717416ebdd0a12e2R1163 

And adjusted the `release` method to use the model delete properly

File: temba/contacts/tests.py [hunk 2]
```diff
@@ -6217,47 +6224,123 @@ def test_read(self):
         self.assertReadFetch(read_url, allow_viewers=True, allow_editors=True, context_object=imp)
 
 
-class PopulateCurrentFlowTest(MigrationTest):
+class FullReleaseDeletedContacts(MigrationTest):
```
Comment (norkans7) [inline temba/contacts/tests.py:6227]: Here I reused the test https://github.com/nyaruka/rapidpro/pull/3689/files#diff-39f3403039114df98053bfecba73c47231022eb2aa652abf6680c2436ba350a4L1447

adjusting the variable with `self.` so I can verify the values after running migration

File: temba/contacts/migrations/0150_full_release_deleted_contacts.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        contact.ticket_events.all().delete()
+
+        for ticket in contact.tickets.all():
+            ticket.broadcasts.update(ticket=None)
+            ticket.delete()
```
Comment (rowanseymour) [inline temba/contacts/migrations/0150_full_release_deleted_contacts.py:16]: let's override `delete()` on ticket and have it clear `self.broadcasts.update(ticket=None)` and then call `super().delete()`

Comment (rowanseymour) [reply temba/contacts/migrations/0150_full_release_deleted_contacts.py:16]: although I guess that doesn't work inside the migration right? But in the real `_full_release` it should work

Comment (norkans7) [reply temba/contacts/migrations/0150_full_release_deleted_contacts.py:16]: Yes that cannot work in the migrations

**Review by rowanseymour — CHANGES_REQUESTED** [review]

**Review by rowanseymour — APPROVED** [review]

**Pull request merged**